File: algorithms/python3/506.py
class Solution:
    def findRelativeRanks(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        nums = sorted([(nums[i], i) for i in range(len(nums))], reverse=True)
        rls = [0] * len(nums)
        if len(nums) >= 1:
            rls[nums[0][1]] = 'Gold Medal'
        if len(nums) >= 2:
            rls[nums[1][1]] = 'Silver Medal'
        if len(nums) >= 3:
            rls[nums[2][1]] = 'Bronze Medal'
        for i in range(3, len(nums)):
            idx, rank = nums[i][1],  i+1
            rls[idx] = str(rank)
        return rls

File: algorithms/rust/506.rs
use std::cmp::Reverse;

impl Solution {
    #[allow(dead_code)]
    pub fn find_relative_ranks(score: Vec<i32>) -> Vec<String> {
        let mut score: Vec<(usize, i32)> = score
            .into_iter()
            .enumerate()
            .map(|(idx, s)| (idx, s))
            .collect();
        score.sort_by_key(|x| Reverse(x.1));

        let mut rslt = vec!["".to_owned(); score.len()];
        for (place, item) in score.into_iter().enumerate() {
            let idx = item.0;
            match place {
                0 => rslt[idx] = "Gold Medal".to_owned(),
                1 => rslt[idx] = "Silver Medal".to_owned(),
                2 => rslt[idx] = "Bronze Medal".to_owned(),
                _ => rslt[idx] = format!("{}", place + 1),
            }
        }
        rslt
    }
}

struct Solution;

fn main() {
    println!("Hello, world!");
}
